File: LC167.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int i = 0, j = numbers.size()-1;

        while(i < j){
            int t = numbers[i] + numbers[j];

            if(t == target)
                return {i+1, j+1};
            if(t < target)
                i++;
            else
                j--;
        }

        return {-1, -1};
    }

};

File: LC167.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i = 0
        j = len(nums)-1

        while i < j:
            t = nums[i] + nums[j]

            if t == target:
                return [i+1, j+1]
            elif t < target:
                i += 1
            else:
                j -= 1
        
        return []
